### ISSUE TITLE
Warnings with Hug component

### ISSUE THREAD

**Issue by mikhailrojo** [priceline/design-system#542]
if add hug component as is from the readme 
```
 <Hug icon="check" text={<Text.span>I hug my props.children</Text.span>}>
      <Card p={4}>Hello</Card>
    </Hug>
```

and open console ther will be a lot of warnings about Icon which cannot be fixed, as no way to pass to Icon the props it askes for.

WARNINGS:

Failed prop type: 'desc' prop supplied to 'Icon' should be a string
 Failed prop type: 'title' prop supplied to 'Icon' should be a string

Example https://n8dyl.codesandbox.io

**Comment by dansternfeld1** [comment]
Hey @mikhailrojo, thank you for reporting this issue.

Yes, you've found a recently-introduced bug in the Hug component.
#509 brought in new Icon propType requirements for accessibility, it may not be null checking properly here: https://github.com/pricelinelabs/design-system/blob/master/packages/icons/src/Icon.js#L69
To address the underlying issue, we should add a null check (`!!props[propName]`) around the `desc` and `title` propType functions in Icon.js

(We should also consider giving the Hug component user a way to add these a11y attributes to their chosen Icon; I'm not sure the extra props are worth the return here, please open a second issue if you're interested in that.)

You're probably seeing a third warning about your `icon` prop as well since it's lowercase.
The example storybook has this error as well:
Warning: Failed prop type: Icon name prop should be uppercase.
Use `ThumbsUp` instead of thumbsUp.
We should clean that up, too, using `lodash.upperfirst` in Hug.js to pass the icon name.

Thanks again for reporting this issue. Sometimes it's hard to see this stuff when you have a "noisy" console -- the more of these we clean up, the easier on the eyes our logging will be.

**Comment by dansternfeld1** [comment]
FYI, this might get addressed with #585

**Comment by James300** [comment]
Just tested and this is fixed.

**Comment by James300** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Warnings with Hug component](https://trello.com/c/nfliHgdS/13-warnings-with-hug-component)
